sale_partner_pricelist/models/res_partner.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Jarsa
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import threading
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerr(models.Model):
+    _inherit = "res.partner"
+
+    allowed_pricelist_ids = fields.Many2many(
+        comodel_name="product.pricelist",
+        copy=False,
+        string="Allowed Pricelists",
+    )
+
+    @api.constrains("property_product_pricelist")
+    def _check_allowed_pricelist(self):
+        test_mode = self.env.registry.in_test_mode() or getattr(
```

Comment:
I don't really like this kind of code. I'd prefer adding a configuration setting to enable the feature.